modules/json_form_widget/src/Element/DateRange.php

```diff
@@ -0,0 +1,208 @@
+<?php
+
+namespace Drupal\json_form_widget\Element;
+
+use Drupal\Core\Datetime\DrupalDateTime;
+use Drupal\Core\Datetime\Element\Datetime;
+use Drupal\Core\Datetime\Entity\DateFormat;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Component\Utility\NestedArray;
+
+/**
+ * Provides a date_range element.
+ *
+ * @FormElement("date_range")
+ * @codeCoverageIgnore
+ */
+class DateRange extends Datetime {
+
+  /**
+   * Get info.
+   */
+  public function getInfo() {
+    $class = get_class($this);
+    $date_format = '';
+    $time_format = '';
+    if (!defined('MAINTENANCE_MODE')) {
+      if ($date_format_entity = DateFormat::load('html_date')) {
+        $date_format = $date_format_entity->getPattern();
+      }
+      if ($time_format_entity = DateFormat::load('html_time')) {
+        $time_format = $time_format_entity->getPattern();
+      }
+    }
+    return $this->getElementInfo($class, $date_format, $time_format);
+  }
+
+  /**
+   * Get element info array.
+   */
+  private function getElementInfo($class, $date_format, $time_format) {
+    return [
+      '#input' => TRUE,
+      '#element_validate' => [
+        [$class, 'validateRange'],
+      ],
+      '#process' => [
+        [$class, 'processDateRange'],
+      ],
+      '#theme' => 'datetime_form',
+      '#theme_wrappers' => ['datetime_wrapper'],
+      '#date_year_range' => '1900:2050',
+      '#date_increment' => 1,
+      '#date_date_callbacks' => [],
+      '#date_timezone' => date_default_timezone_get(),
+      '#date_date_element' => 'date',
+      '#date_date_format' => $date_format,
+      '#date_time_element' => 'time',
+      '#date_time_format' => $time_format,
+    ];
+  }
+
+  /**
+   * Callback for getting the value of the date range element.
+   */
+  public static function valueCallback(&$element, $input, FormStateInterface $form_state) {
+    $element += ['#date_timezone' => date_default_timezone_get()];
+    if ($input !== FALSE) {
```

Comment:
I'd just return `if ($input == FALSE) return $input;` in the first line instead of putting the entire method under this condition.